internal/server/v1/warden/service.go

```diff
@@ -0,0 +1,167 @@
+package warden
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	shieldv1beta1rpc "buf.build/gen/go/gotocompany/proton/grpc/go/gotocompany/shield/v1beta1/shieldv1beta1grpc"
+	shieldv1beta1 "buf.build/gen/go/gotocompany/proton/protocolbuffers/go/gotocompany/shield/v1beta1"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+//go:generate mockery --with-expecter --keeptree --case snake --name Doer
+
+const (
+	baseURL = "https://go-cloud.golabs.io"
+)
+
+type Service struct {
+	shieldClient shieldv1beta1rpc.ShieldServiceClient
+	doer         HTTPClient
+}
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+func NewService(shieldClient shieldv1beta1rpc.ShieldServiceClient, doer HTTPClient) *Service {
+	return &Service{
+		shieldClient: shieldClient,
+		doer:         doer,
+	}
+}
+
+func (c *Service) TeamList(ctx context.Context, userEmail string) (*TeamData, error) {
+	endpoint := "/api/v1"
+	userPath := "/users/"
+	teamsEndpoint := "/teams"
+
+	url := baseURL + endpoint + userPath + userEmail + teamsEndpoint
```

Comment:
warden is a purely http thing, so we are resolving this via service. We could have used it as client if it was gRPC.

Reply:
How is grpc a dependency for a client. 
it's good to generate a client since it's an external service plus has swagger available

Reply:
I confirmed with other peer members also, the service implementation of warden seems suffice the need looking the scope of story.

if you feel warden client should be created, we can create another story for this will get this implemented.